### DISCUSSION TITLE
EK plaintext cpp

### DISCUSSION THREAD

**Discussion by gino-1557** [microsoft/TSS.MSR#137] [Q&A]
hello,
I would like to have the EK in plaintext in the TSS.CPP project. I have found in the samples an approach how it could work.

https://github.com/microsoft/TSS.MSR/blob/master/TSS.CPP/Samples/Samples.cpp
void Samples::Activate()
{
    Announce("ActivateCredential");

    // Make a new EK and get the public key
    TPM_HANDLE ekHandle = MakeEndorsementKey();
    auto ekPubX = tpm.ReadPublic(ekHandle);
    TPMT_PUBLIC& ekPub = ekPubX.outPublic;
}

I am still struggling to get out the public part of EK in plaintext. 
About help I would be happy. Thanks.

**Comment by uurturan** [comment]
`ekPubX.outPublic.ToString()` should return the public key (among a few other things) in "plain text". You could also check out
`ekPubX.outPublic.unique->toBytes()` which should return the public key as a byte array.
